Config file for edition picker

diogoalexandrefranco opened the pull request:
This is a first try of an example of the use of a config file (src/app/config.json) to decide which components to show in the new sinfo edition picker. This is related to issue #21.

For the example, I also created dummy *Team*, *Speakers* and *Companies* components.

Explanation:
*EditionPicker* renders a few buttons and the child components. Its state maintains which edition is picked, and this is passed through props to the *Team*, *Speakers* and *Companies* components. These decide whether to render by reading the config file.

If this is not idiomatic within redux or whatever, please show me the ways!

johnytiago commented on src/app/components/EditionPicker.js:
```diff
@@ -0,0 +1,30 @@
+  render () {
+    return (
+      <div className='Edition-picker'>
+        <Button onClick={() => this.setState({edition: 'sinfo21'})}>SINFO 21</Button>
```
I think you can concatenate all this 4 lines into one and just use a function that takes the argument `SINFO 2X`, then you just have to lowercase it, and remove the whitespace (`.replace(/\s/g,'')`). Then you just have to call `onClick = functionName("SINFO 2X").bind(this)` or something like that. Am I right?

JGAntunes replied:
I think just for the purposes used here (examples) we can leave it as it is, there's no point in over engineering this.

johnytiago commented on src/app/components/Speakers.js:
```diff
@@ -0,0 +1,17 @@
+  render () {
+    return (
+      <div>
+        { this.shouldRender(this.props.edition) && <h3> The speakers component is rendered</h3> }
```
can't this evaluating be done on the parent component. This component will always render, you are just specifying if it will be empty or not, right?

diogoalexandrefranco replied:
Yes, I think it can. I discussed it with @JGAntunes, we were not 100% sure which way is better but we concluded yours might be simpler. I'll update the PR.

JGAntunes commented:
@diogoalexandrefranco can you check @johnytiago's comments and resolve the conflicts so we can merge this?

diogoalexandrefranco commented:
I incorporated @johnytiago suggestion of handling the render logic in the edition picker component, and rebased this PR to incorporate the new changes in the react branch. The rendering of the Speakers and Members is now handled inside EditionPicker, since they might not always be visible.

If there is an issue with clicking the example buttons, I think it has to do with the carousel taking more space than that which is visible. I can look into that further.
